lib/FusionInventory/Agent/Task/Inventory/Win32/Firewall.pm

```diff
@@ -0,0 +1,201 @@
+                signaturesKey => $signaturesKey
+            );
+        }
+
+        next unless $profile;
+	
+        my $category = hex2dec($profile->{'/Category'});
+        if (not defined $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}) {
+            $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS} = [];
+        }
+        my $connection = {DESCRIPTION => $interface->{DESCRIPTION}};
+        $connection->{IPADDRESS} = $interface->{IPADDRESS} if ($interface->{IPADDRESS});
+        $connection->{IPADDRESS6} = $interface->{IPADDRESS6} if ($interface->{IPADDRESS6});
+        push @{$params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}}, $connection;
+    }
+
+    my @profiles = ();
+    for my $p (values %{$params{firewallProfiles}}) {
+        my @p;
+        if ($p->{CONNECTIONS} && ref($p->{CONNECTIONS}) eq 'ARRAY') {
+            my @conns = @{$p->{CONNECTIONS}};
+            delete $p->{CONNECTIONS};
+            for my $conn (@conns) {
+                my $newP = dclone $p;
+                while (my ($k, $v) = each %$conn) {
+                    $newP->{$k} = $v;
+                }
+                push @p, $newP;
+            }
+        } else {
+            push @p, $p;
+        }
+        push @profiles, @p;
+    }
+
+    return @profiles;
+}
+
+sub _getConnectionDomainSettings {
+    my (%params) = @_;
+
+    return unless $params{guid};
+
+    my $registeredAdapter = getRegistryKey(
+        logger => $params{logger},
+        path => 'HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/services/Tcpip/Parameters/DNSRegisteredAdapters/' . $params{guid}
+    );
+    if ($registeredAdapter && $registeredAdapter->{'/PrimaryDomainName'}) {
+        return $registeredAdapter;
+    } else {
+        return;
+    }
+}
+
+sub _retrieveFirewallProfileWithoutDomain {
+    my (%params) = @_;
+
+    return unless $params{DNSDomain} && $params{profilesKey} && $params{signaturesKey};
+
+    my $profilesKey = $params{profilesKey};
+    my $signaturesKey = $params{signaturesKey};
+
+    my $dnsDomain = $params{DNSDomain};
+    my $profileGuid;
+    for my $sig (values %{$signaturesKey->{'Managed/'}}, values %{$signaturesKey->{'Unmanaged/'}}) {
+        if ($sig->{'/FirstNetwork'} eq $dnsDomain) {
+            $profileGuid = $sig->{'/ProfileGuid'};
+            last;
+        }
+    }
+    return unless $profileGuid && $profilesKey->{$profileGuid};
+
+    return $profilesKey->{$profileGuid};
+}
+
+sub _retrieveProfilesAndSignaturesKey {
+    my (%params) = @_;
+
+    my $networkListKey = getRegistryKey(
+        logger => $params{logger} || undef,
+        path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows NT/CurrentVersion/NetworkList'
+    );
+    return unless $networkListKey;
+
+    if ($networkListKey->{'Profiles/'} && $networkListKey->{'Signatures/'}) {
+        return ($networkListKey->{'Profiles/'}, $networkListKey->{'Signatures/'});
+    } else {
```

Comment:
Again, I don't see the need of a "else { return; }" as last case at the end of a method, or clearly return a default value without the "else"